konnectivity-client/pkg/client/client_test.go

```diff
@@ -70,10 +73,10 @@ func TestData(t *testing.T) {
 		conns:       make(map[int64]*conn),
 	}
 
-	go tunnel.serve(&fakeConn{})
+	go tunnel.serve(ctx, &fakeConn{})
```

Comment:
Would be nice if we had a test which used a fakeDelayConn so we could add a test for timeout on the context.